electrumutils.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import binascii
+from utils import BITCOIND_CONFIG
+import time
+import sys
+import os
+import tempfile
+import traceback
+import asyncio
+from electrumx.server.controller import Controller
+from electrumx.server.env import Env
+import threading
+from electrum.daemon import Daemon, get_fd_or_server
+from electrum import simple_config, constants, keystore, util
+from electrum.wallet import Wallet
+from electrum.storage import WalletStorage
+from electrum.mnemonic import Mnemonic
+
+bh2u = lambda x: binascii.hexlify(x).decode('ascii')
+
+class ElectrumX:
+    def __init__(self):
+        print("server/__init__")
+        self.loop = asyncio.new_event_loop()
+        self.evt = asyncio.Event(loop=self.loop)
+
+    def kill(self):
+        print("server/kill")
+        async def set_evt():
+            self.evt.set()
+        asyncio.run_coroutine_threadsafe(set_evt(), self.loop)
+
+    def start(self):
+        print("server/start")
+        from os import environ
+        rpc_port = 8000
+        os.system("rm -rf COIN hist meta utxo")
+        environ.update({
+            "COIN": "BitcoinSegwit",
+            "SSL_KEYFILE": "/home/janus/electrumx/key.pem",
+            "SSL_CERTFILE": "/home/janus/electrumx/cert.pem",
+            "TCP_PORT": "51001",
+            "SSL_PORT": "51002",
+            "RPC_PORT": str(8000),
+            "NET": "regtest",
+            "DAEMON_URL": "http://rpcuser:rpcpass@127.0.0.1:" + str(BITCOIND_CONFIG.get('rpcport', 18332)),
+            "DB_DIRECTORY": "."
+        })
+        def target():
+            loop = self.loop
+            asyncio.set_event_loop(loop)
+            env = Env()
+            env.loop_policy = asyncio.get_event_loop_policy()
+            self.controller = Controller(env)
+            logging.basicConfig(level=logging.DEBUG)
+            loop.run_until_complete(asyncio.wait([self.controller.serve(self.evt), self.evt.wait()], return_when=asyncio.FIRST_COMPLETED))
+            loop.close()
+
+        self.thread = threading.Thread(target=target)
+        self.thread.start()
+        #controller.run()
+
+    def wait(self):
+        print("server/wait")
+        self.thread.join()
+
+class ElectrumDaemon:
+    def __init__(self, electrumx):
+        self.electrumx = electrumx
+        self.port = -1
+        self.actual = None
+
+    def start(self):
+        print("daemon/start")
+        user_config = {
+            "auto_connect": False,
+            "oneserver": True,
+            "server": "localhost:51001:t",
+            "request_initial_sync": False,
+        }
+        user_dir = tempfile.mkdtemp(prefix="lightning-integration-electrum-")
```

Comment:
The fixtures now contain a fixture called directory which is just a path. That directory is test-specific and should be used as the root for all things related to that test. Could you use that instead, so we can clean up tests that pass correctly?